Custom Jest matchers causing issues with Jest upgrade

We're unable to upgrade jest because the custom jest matcher types in EUI are conflicting with the version of Jest we're trying to install. I don't think there's a good reason for these extensions personally, and would prefer if they could just be implemented as helper functions so that the types could be removed. Thoughts?

Kibana Jest upgrade: https://github.com/elastic/kibana/pull/53079
Error from https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15622/execution/node/128/log/
```
03:42:42  kibana: node_modules/@elastic/eui/eui.d.ts(5656,20): error TS2428: All declarations of 'Matchers' must have identical type parameters.
03:42:42  kibana: node_modules/@types/jest/index.d.ts(729,15): error TS2428: All declarations of 'Matchers' must have identical type parameters.
```

Problematic module:
```ts
declare module '@elastic/eui/src/components/datagrid/data_grid.test' {
	 global {
	    namespace jest {
	        interface Matchers<R> {
	            toBeEuiPopover(): R;
	            euiPopoverToBeOpen(): R;
	        }
	    }
	}
	export {};
}
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,10 @@
 
 - Reverted docs changes in `17.2.0` that caused the build script to die ([#2672](https://github.com/elastic/eui/pull/2672))
 
+**Bug fixes**
+
+- Removed TypeScript definitions in `*.test.tsx?` files from _eui.d.ts_ ([#2673](https://github.com/elastic/eui/pull/2673))
+
 ## [`17.2.0`](https://github.com/elastic/eui/tree/v17.2.0)
 
 - Improved a11y of `EuiNavDrawer` lock button state via `aria-pressed` ([#2643](https://github.com/elastic/eui/pull/2643))

--- a/scripts/dtsgenerator.js
+++ b/scripts/dtsgenerator.js
@@ -32,6 +32,8 @@ const generator = dtsGenerator({
     'node_modules/**/*.d.ts',
     '*/custom_typings/**/*.d.ts',
     'src-framer/**/*',
+    '**/*.test.ts',
+    '**/*.test.tsx',
   ],
   resolveModuleId(params) {
     if (

